lib/internal/fs/promises.js

```diff
@@ -433,6 +439,15 @@ async function access(path, mode = F_OK) {
                         kUsePromises);
 }
 
+async function cp(src, dest, options) {
+  options = validateCpOptions(options);
+  src = getValidatedPath(src, 'src');
+  dest = getValidatedPath(dest, 'dest');
+  src = pathModule.toNamespacedPath(src);
+  dest = pathModule.toNamespacedPath(dest);
```

Comment:
```suggestion
function cp(src, dest, options) {
  options = validateCpOptions(options);
  src = pathModule.toNamespacedPath(getValidatedPath(src, 'src'));
  dest = pathModule.toNamespacedPath(getValidatedPath(dest, 'dest'));
```
I think this is more readable than having src,dest,src,dest.

Reply:
`async` is to make sure that errors thrown by validation end up in rejections and not sync exceptions.

Reply:
Thanks for the suggestion 👍